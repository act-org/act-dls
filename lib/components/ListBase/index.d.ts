/**
 * @prettier
 */
import * as React from 'react';
import { ListProps } from '@material-ui/core';
export declare type Props = ListProps;
declare const ListBase: React.FC<Props>;
export default ListBase;
//# sourceMappingURL=index.d.ts.map