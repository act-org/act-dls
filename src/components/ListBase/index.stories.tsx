/**
 * @prettier
 */

import * as React from 'react';
import { boolean } from '@storybook/addon-knobs';

import ListBase from '.';

export const Base = (): React.ReactElement<any> => (
  <ListBase
    dense={boolean('Dense?', false)}
    disablePadding={boolean('Disable Padding?', false)}
  >
    List
  </ListBase>
);

export default {
  component: ListBase,
  parameters: {
    info: {
      text: 'This is the base variant of the List component.',
    },
  },
  title: 'Atoms/List',
};
