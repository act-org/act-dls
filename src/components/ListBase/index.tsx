/**
 * @prettier
 */

import * as React from 'react';
import { List, ListProps } from '@material-ui/core';

export type Props = ListProps;

const ListBase: React.FC<Props> = (props: Props): React.ReactElement<any> => (
  <List {...props} />
);

export default ListBase;
