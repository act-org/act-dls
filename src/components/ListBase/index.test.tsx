/**
 * @prettier
 */

import * as React from 'react';

import render from '~/helpers/test/render';

import ListBase from '.';

describe('ListBase', () => {
  it('matches the snapshot', () => {
    const { container } = render(<ListBase dense>List</ListBase>);

    expect(container).toMatchSnapshot();
  });
});
