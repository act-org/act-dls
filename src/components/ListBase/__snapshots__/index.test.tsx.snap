// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListBase matches the snapshot 1`] = `
<div>
  <ul
    class="MuiList-root MuiList-dense MuiList-padding"
  >
    List
  </ul>
</div>
`;
